workflow/templateresolution/context_test.go

```diff
@@ -339,6 +355,22 @@ func TestResolveTemplate(t *testing.T) {
 	assert.Equal(t, "whalesay-with-arguments", tmpl.Name)
 	assert.Equal(t, []string{"{{inputs.parameters.message}}-foo"}, tmpl.Container.Args)
 
+	// Get the template of template reference with passthrough arguments.
```

Comment:
I think this validation does nothing for `passthroughRegexp` handling but just does the same template resolving as the others. I realized we need `passthroughRegexp` handling in `MergeReferredTemplate` and we can write tests for that function instead.

Reply:
Agreed, am resolving.